src/survey/surveyObj.ts

```diff
@@ -0,0 +1,295 @@
+import { AuthGroup } from '../auth'
+import { SystemError } from '../error'
+import {
+  NodeDefBooleanProps,
+  NodeDefCodeProps,
+  NodeDefDecimalProps,
+  NodeDefFileProps,
+  NodeDefMeta,
+  NodeDefProps,
+  NodeDefPropsAdvanced,
+  NodeDefTaxonProps,
+  NodeDefTextProps,
+  NodeDefType,
+} from '../nodeDef'
+import { Survey, SurveyProps } from './survey'
+
+export abstract class ArenaObj<P> {
+  uuid: string
+  id?: number
+  props?: P
+
+  constructor(uuid: string) {
+    this.uuid = uuid
+  }
+}
+
+export abstract class NodeDefObj<P extends NodeDefProps = NodeDefProps> extends ArenaObj<P> {
+  protected _survey: SurveyObj
+  protected _parent?: EntityDefObj
+
+  analysis?: boolean
+  dateCreated?: string
+  dateModified?: string
+  deleted?: boolean
+  draft?: boolean
+  meta?: NodeDefMeta
+  propsAdvanced?: NodeDefPropsAdvanced
+  propsAdvancedDraft?: NodeDefPropsAdvanced
+  propsDraft?: P
```

Comment:
P ?

Reply:
P extends NodeDefProps is the generic for the props